src/model/SearchDirectory.cpp

```diff
@@ -1,36 +1,45 @@
 #include "SearchDirectory.h"
 #include <iostream>
-vector<pair<int, DreeNode *>> SearchDirectory::search(DreeNode *root, string &query, DreeHelpersI *dreeHelpers) {
-    vector<pair<int, DreeNode *>> searchResult;
+vector<pair<int, pair<string, string>>> SearchDirectory::search(DreeNode *root, string &query, DreeHelpersI *dreeHelpers) {
```

Comment:
Isn't it decreasing the readability of the function?

Reply:
Do you want the function signature to remain as - vector<pair<int, DreeNode *>> @ujjwall-R